nsv13/game/lazylift.dm

```diff
@@ -0,0 +1,394 @@
+			sound_effect(start=FALSE)
+		move_platform(_deck)
+	platform_location.unbolt_doors()
+	addtimer(VARSET_CALLBACK(src, in_use, FALSE), wait_time)
+
+/obj/machinery/lazylift/master/proc/move_platform(targetDeck)
+	var/obj/machinery/lazylift/target = null
+	for(var/obj/machinery/lazylift/L in decks) //First pass: Find the target
+		if(L.deck == targetDeck)
+			target = L
+			for(var/turf/T in target.platform)
+				T.ChangeTurf(turf_type)
+			if(target == src) //Bottom floor, time to karmic anyone who was stupid enough as to stand here...
+				for(var/turf/T in platform)
+					for(var/atom/movable/AM in T)
+						if(isturf(AM))
+							continue
+						if(isliving(AM))
+							var/mob/living/karmics_victim = AM //SQUISH
+							karmics_victim.gib() //Elevator suicides are...a thing I guess!
+						else
+							AM.ex_act(4) //Crush
+			continue
+	if(!target)
+		message_admins("Turbolift couldnt find a target..this is bad...")
+		return FALSE
+	for(var/turf/T in platform_location.platform)
+		for(var/atom/movable/AM in T.contents)
+			if(AM.type in moving_blacklist) //To stop the lift moving itself
+				continue
+			AM.forceMove(get_turf(locate(AM.x, AM.y, target.z)))
+	for(var/obj/machinery/lazylift/next in decks) //Second pass: Turn all the other lift platforms into openspace. I don't hugely care if things fall through, that's a feature.
+		if(next == target)
+			continue
+		for(var/turf/open/T in next.platform)
+			var/obj/machinery/door/airlock/turbolift_door = locate(/obj/machinery/door/airlock) in T
+			if(!turbolift_door)
+				if(next != src)
+					T.ChangeTurf(/turf/open/openspace)
+				else
+					if(!istype(T, /turf/open/floor/plasteel/elevatorshaft))
+						T.ChangeTurf(/turf/open/floor/plasteel/elevatorshaft, list(/turf/open/openspace, /turf/open/floor/plating))
+						T.icon_state = "elevatorshaft"
+	platform_location = target
+
+//Special FX and stuff.
+
+/obj/machinery/lazylift/master/proc/sound_effect(start)
+	if(start)
+		for(var/mob/M in get_area(src))
+			SEND_SOUND(M, 'nsv13/sound/effects/lift/lift_start.ogg')
+			shake_camera(M, 2, 1)
+			if(!isliving(M))
+				continue
+			if(obj_flags & EMAGGED)
+				var/mob/living/karmics_victim = M
+				karmics_victim.Knockdown(5 SECONDS)
+				shake_camera(karmics_victim, 10, 1)
+				to_chat(karmics_victim, "<span class='warning'>You're pressed into the floor as the lift rapidly accelerates!</span>")
+				if(prob(50)) //Unlucky fucker
+					to_chat(karmics_victim, "<span class='warning'>You hit your head as you're thrown about wildly!</span>")
+					karmics_victim.adjustBrainLoss(10)
+					playsound(karmics_victim.loc, 'nsv13/sound/effects/lift/bonk.ogg', 100)
+				if(HAS_TRAIT(karmics_victim, TRAIT_SEASICK)) //oh my god you poor soul
+					to_chat(karmics_victim, "<span class='warning'>You can feel your head start to swim...</span>")
+					if(prob(40))
+						karmics_victim.adjust_disgust(20)
+					else
+						karmics_victim.adjust_disgust(10)
+		sleep(1.4 SECONDS) //Sound time!
```

Comment:
This is potentially a unsuitable sleep here have you thought about using a callback instead?

Reply:
Potentially, but sleeps are fine in this instance, as I've heard from multiple people. As long as they're not excessively long, they're on par with timers. This is the intended use case for them (short, animation related breaks). If I were to use timers, it'd also increase proc call overhead.